### ISSUE TITLE
Using Dict for potential increase in configuration settings

### ISSUE THREAD

**Issue by peter1000** [MichaelHatherly/Lexicon.jl#52]
this is taken from a comment by MichaelHatherly in: https://github.com/MichaelHatherly/Lexicon.jl/pull/50

I think we're going to need a Dict{Symbol, Any}, or something, with the potential increase in configuration settings.
```
function save(file::String, modulename::Module, plmain, config::Dict{Symbol, Any})
    mime = MIME("text/$(strip(last(splitext(file)), '.'))")
    save(file, mime, documentation(modulename), plmain, config)
end
```
Not with this PR though, we can change it later on.

**Comment by peter1000** [comment]
I will do that first if you don't mind. Wouldn't it be better to use instead of a Dict a Type toherwise we have to check for valid keys ect.

**Comment by MichaelHatherly** [comment]
That's fine. Could it be a separate PR though, just so we keep changes relatively self-contained?

**Comment by MichaelHatherly** [comment]
A potential sketch using a `Config` object rather than a `Dict`:

```julia
type Config

    include_internal :: Bool
    mathjax          :: Bool

    const fields   = fieldnames(Settings)
    const defaults = Dict{Symbol, Any}(
        :include_internal => true,
        :mathjax          => false
        )

    function Config(; args...)
        this = new()
        for (k, v) in merge(defaults, Dict(args))
            k ∈ fields ?
                setfield!(this, k, v) :
                warn("Invalid setting: '$(k) = $(v)'.")
        end
        this
    end

end
```

Then we can reference settings using `config.mathjax` rather than `config[:mathjax]`, which I feel is a much nicer way to write it.

**Comment by peter1000** [comment]
I can do that.
How would you suggest that a user uses it.
```
conf = Config()
config.mathjax = true
save("docs/api/Lexicon.md", Lexicon, config)
```

**Comment by MichaelHatherly** [comment]
`Config` should, for the moment, be internal. Changing the `save` function to accept arbitrary keyword args and build the `Config` object internally should work fine using the "double splat" pattern:

```julia
function save(name, module; args...)
    config = Config(; args...)
    # ...
end
```

or something like that.

**Comment by peter1000** [comment]
Ok

**Comment by MichaelHatherly** [comment]
Should mean that it's not a breaking change to the interface.

**Comment by peter1000** [comment]
just a coding question - so I learn something: your example above included this.

```
        for (k, v) in merge(defaults, Dict(args))
            k ∈ fields ?
                setfield!(this, k, v) :
                warn("Invalid setting: '$(k) = $(v)'.")
        end
```

Question: what does the **∈** mean - is it the same like saying `k in fields ?` or does it mean something else in julia? Thanks

**Comment by MichaelHatherly** [comment]
Yeah, it just means `in`. In fact I think it's probably just defined as `const ∈ = in` somewhere in `Base`. Inputting it is just `\in<tab>` so it's not much longer either. I find it more noticeable when reading. Doesn't work in `for` loops though.

**Comment by MichaelHatherly** [comment]
Closed by #56.
